scripts/clan_resources/herb/herb_supply.py

```diff
@@ -0,0 +1,726 @@
+                total_herb_amount = self.get_single_herb_total(herb_used)
+
+                amount_used = randint(1, total_herb_amount if total_herb_amount < 4 else 4)
+                strength = 1
+                for level, herb_list in source_dict[name]["herbs"].items():
+                    if herb_used in herb_list:
+                        strength = int(level)
+
+                self.remove_herb(herb_used, amount_used)
+
+                self.__apply_herb_effect(treatment_cat, name, herb_used, chosen_effect, amount_used, strength)
+
+            else:
+                self.__apply_lack_of_herb(treatment_cat, name, chosen_effect)
+
+    def _gather_herbs(self, med_cat):
+        """
+        finds out what herbs an individual med cat gathered during moon skip and adds those herbs to collection
+        and log
+        """
+
+        list_of_herb_strs, found_herbs = self.get_found_herbs(med_cat)
+
+        if found_herbs:
+            # add found herbs to log
+            self.log.append(i18n.t(
+                "screens.med_den.log_messages.gather_success",
+                name=med_cat.name,
+                herbs=adjust_list_text(list_of_herb_strs)
+            ))
+        else:
+            self.log.append(i18n.t(
+                "screens.med_den.log_messages.gather_fail",
+                name=med_cat.name
+            ))
+
+    def _remove_from_storage(self, herb: str, needed_num: int) -> int:
+        """
+        removes needed_num of given herb from storage until needed_num is met or storage is empty, if storage runs out
+        before needed_num is met, returns excess
+        """
+        while needed_num > 0 and self.storage[herb]:
+            # remove from oldest stock
+            self.storage[herb][-1] -= needed_num
+            # if that stock runs out, move to next oldest stock
+            if self.storage[herb][-1] < 0:
+                needed_num = abs(self.storage[herb][-1])
+                self.storage[herb].pop(-1)
+
+        return needed_num
+
+    def __apply_herb_effect(self, treated_cat, condition: str, herb_used: str, effect: str, amount_used: int,
+                            strength: int):
+        """
+        applies the given effect to the treated_cat
+        """
+
+        # grab the correct condition dict so that we can modify it
+        if condition in treated_cat.illnesses:
+            con_info = treated_cat.illnesses[condition]
+        elif condition in treated_cat.injuries:
+            con_info = treated_cat.injuries[condition]
+        else:
+            con_info = treated_cat.permanent_condition[condition]
+
+        amt_modifier = int(amount_used * .5) if int(amount_used * .5) >= 1 else 1
+
+        effect_message = ""
+        # apply mortality effect
+        if effect == HerbEffect.MORTALITY:
+            con_info[effect] += (
+                    3 * strength + amt_modifier
+            )
+            effect_message = i18n.t("screens.med_den.log_messages.mortality_down")
```

Comment:
These don't point to valid i18n strings i don't think?

Reply:
You sure?  They show up fine in testing.